### PR TITLE
add durian.js config to use multi cores

### PR DESCRIPTION
**Pull request by mario-huang**

### CODE REVIEW

**Review by waghanza — COMMENTED** [review]
Nice, thanks for this.

Could you then remove cluster.ts from all frameworks with bun then ?

**Comment by mario-huang** [comment]
> Nice, thanks for this.
> 
> Could you then remove cluster.ts from all frameworks with bun then ?

OK, I will try.

**Comment by mario-huang** [comment]
@waghanza It seems that bunicorn has bug in itself, what should I do.

File: javascript/0http-bun/config.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@ framework:
   github: BackendStack21/0http-bun
   version: 1.0
   engines:
-    - bun
+    - durian.js
```
Comment (waghanza) [inline javascript/0http-bun/config.yaml:5]: I do not understand this, the engine is still bun

File: javascript/config.yaml [hunk 2]
```diff
@@ -42,6 +42,13 @@ language:
       bootstrap:
         - npm install
       command: node app.js
+    durian.js:
```
Comment (waghanza) [inline javascript/config.yaml:45]: You can drop the old config for bun

**Review by waghanza — CHANGES_REQUESTED** [review]

File: javascript/config.yaml (hunk 2 above)
Comment (mario-huang) [reply javascript/config.yaml:45]: OK, got it.

**Comment by mario-huang** [comment]
When I test, I found there is still bug with pm2 and bun. So no rush, I will keep eyes on it.
https://github.com/Unitech/pm2/issues/5893

**Review by waghanza — APPROVED** [review]

**Comment by waghanza** [comment]
Thanks, I'll merge and fix it

**Comment by mario-huang** [comment]
@waghanza Hi.
I think perhaps it's better not to merge for now. During my search over the past few days, I found that PM2 still doesn't support Bun running in cluster mode at the moment. If it's run in fork mode, the program will throw an error due to port conflicts.
This made me consider another issue: as a benchmark, wouldn't it be better not to introduce additional load balancing like pm2? After all, different load balancing implementations can also affect the results. If we let all frameworks run on a single CPU core, we can maintain the simplicity of the frameworks and better reflect the performance of the frameworks themselves.

**Comment by waghanza** [comment]
Understood @mario-huang. We actually only display one engine, per frameworks, I'll be too heavy to maintain to have multiple load balancesrs. Seems like pm2 is the most used in javascript world, but what load balance do you think about ?

**Comment by mario-huang** [comment]
Yes, pm2 is the most used in javascript world, it let nodejs apps use multi CPU cores. 
With the widespread adoption of Kubernetes (k8s), PM2 is no longer essential. We can directly utilize k8s' load balancing features to achieve the same results. Alright, back to the JavaScript situation in this project. Currently, we have Node.js, Deno, and Bun. Both Node.js and Deno use PM2 to take advantage of multi-core CPUs and share port 3000. The Bun project directly runs Bun.serve({reusePort: true}), achieving the same goal. However, Durian.js does not run based on Bun.serve(), which prevents it from having multiple instances and sharing ports like other Bun projects. I tried using PM2 to solve this problem, but unfortunately, it doesn't seem feasible at the moment. So, it looks like I'll need to wait for PM2 to support Bun, or add cpu: "1.0" in all the Dockerfiles to ensure that all frameworks use only one core. This way, we can compare performance on a single core, and none of the frameworks would need any load balancing.

**Comment by waghanza** [comment]
Ok. Understood. So I'll restore the old situation, and let you make a PR when this feature will be available on pm2, it is not doable to use k8s here with multiple nodes, at least with only one machine because this runs on my own infrastructure (not cloud).

Thanks for your work on this btw :heart:

**Pull request merged**